### PR TITLE
Minor compilation improvements

### PR DESCRIPTION
**Pull request by hubertp**
### Pull Request Description

Drop `Core` implementation (replacement for IR) as it (sadly) looks increasingly
unlikely this effort will be continued. Also, it heavily relies
on implicits which increases some compilation time (~1sec from `clean`)

Related to https://www.pivotaltracker.com/story/show/182359029

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1380,7 +1380,6 @@ lazy val `engine-runner` = project
   .dependsOn(pkg)
   .dependsOn(cli)
   .dependsOn(`library-manager`)
-  .dependsOn(`language-server`)
```
Comment (4e6) [inline build.sbt]: I'm pretty sure we need this dependency or it won't compile :thinking:

**Review by 4e6 — APPROVED** [review]
The best type of PR

File: build.sbt (hunk 1 above)
Comment (hubertp) [reply build.sbt]: Yeah, looks like runner is dependent on it.

**Review by JaroslavTulach — APPROVED** [review]

File: build.sbt [hunk 2]
```diff
@@ -1222,15 +1197,12 @@ lazy val runtime = (project in file("engine/runtime"))
       "com.novocode"        % "junit-interface"       % "0.11"            % Test exclude ("junit", "junit-dep")
     ),
     // Note [Unmanaged Classpath]
-    Compile / unmanagedClasspath += (`core-definition` / Compile / packageBin).value,
-    Test / unmanagedClasspath += (`core-definition` / Compile / packageBin).value,
```
Comment (JaroslavTulach) [inline build.sbt:1226]: Less dependencies of `engine/runtime` is good!

**Pull request merged**